src/App.tsx

```diff
@@ -17,18 +60,27 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter values={todoFilter} set={setTodoFilter} />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loader && <Loader />}
+              <TodoList
+                todos={currentTodos()}
+                setSelectedTodo={setSelectedTodo}
+                selectedTodo={selectedTodo}
+              />
```

Comment:
Ensure that the `TodoList` component uses a unique `key` prop for each list item. This is crucial to avoid rendering issues.